internal/backends/database.go

```diff
@@ -109,6 +124,10 @@ type CreateCollectionParams struct {
 func (dbc *databaseContract) CreateCollection(ctx context.Context, params *CreateCollectionParams) error {
 	defer observability.FuncCall(ctx)()
 
+	must.BeTrue(params.CappedSize >= 0)
+	must.BeTrue(params.CappedSize%256 == 0)
```

Comment:
By the way 256 increment specific to Oplog only? For normal capped collection is this requirement too strict? :thinking:

Reply:
That restriction is documented for all collections: https://www.mongodb.com/docs/manual/core/capped-collections/#create-a-capped-collection

Reply:
Thanks :pray: So before reaching this point, the size will be rounded :+1: